src/quill.coffee

```diff
@@ -241,6 +253,36 @@ class Quill extends EventEmitter2
     params[3] ?= Quill.sources.API
     return params
 
+  _getLeafFormats: (start, end) ->
+    if start == end
+      [line, offset] = @editor.doc.findLineAt(start)
+      if offset == 0
+        contents = this.getContents(start, end + 1)
+      else
+        contents = this.getContents(start - 1, end)
+    else
+      contents = this.getContents(start, end)
+    formats = _.map(contents.ops, 'attributes')
+    return this._intersectFormats(formats)
+
+  _getLineFormats: (start, end) ->
+    formats = []
+    [firstLine, offset] = @editor.doc.findLineAt(start)
+    [lastLine, offset] = @editor.doc.findLineAt(end)
+    currLine = firstLine
+    while true
+      formats.push(_.clone(currLine.formats))
+      break if currLine == lastLine
+      currLine = currLine.next
+    return this._intersectFormats(formats)
+
+  _intersectFormats: (formatGroups) ->
+    return _.reduce(formatGroups.slice(1), (inAll, formats) ->
```

Comment:
@pedrosanta if you look at line 284, the first leaf is passed in there. This is because the `reduce` function requires a starting point, so the first leaf it broken off and used as the base and then all subsequent leaves are compared on top of it.